src/mongo_odm/nvp.hpp

```diff
@@ -21,6 +21,24 @@
 #include <tuple>
 #include <type_traits>
 
+#include <bsoncxx/types.hpp>
+
+#include <mongo_odm/util.hpp>
+
+// Preprocessor templates for manipulating multiple arguments.
+#define PP_NARG(...) PP_NARG_(__VA_ARGS__, PP_RSEQ_N())
```

Comment:
For all these `PP_*` macros, are they being used outside of this file? if not, surround this with push macro / undef / pop macro so we don't expose these. (see the libmongocxx prelude/postlude for good examples of this)